src/main/java/domain/Board.java

```diff
@@ -0,0 +1,77 @@
+package domain;
+
+import domain.chessPiece.ChessPiece;
+import domain.hurdlePolicy.HurdlePolicy;
+import domain.path.Path;
+import domain.position.ChessPiecePositions;
+import domain.position.ChessPosition;
+import domain.score.Score;
+import domain.type.ChessTeam;
+
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+
+public class Board {
+
+    private final ChessPiecePositions chessPiecePositions;
+    private final Map<ChessTeam, Score> scores = new EnumMap<>(ChessTeam.class);
+
+    public Board(final ChessPiecePositions chessPiecePositions) {
+        this.chessPiecePositions = chessPiecePositions;
+        scores.putAll(Map.of(
+                ChessTeam.RED, Score.zero(),
+                ChessTeam.BLUE, Score.zero()
+        ));
+    }
+
+    public boolean isExistPieceAt(ChessPosition position) {
+        return chessPiecePositions.existChessPieceByPosition(position);
+    }
+
+    public void move(final ChessTeam currentTeam, final ChessPosition from, final ChessPosition to) {
+        validateTeam(currentTeam, from);
+        validateDestination(from, to);
+        if (chessPiecePositions.existChessPieceByPosition(to)) {
+            killTarget(currentTeam, to);
+        }
+        chessPiecePositions.move(from, to);
+    }
+
+    public List<ChessPosition> getAvailableDestination(final ChessPosition position) {
+        ChessPiece chessPiece = chessPiecePositions.getChessPieceByPosition(position);
+        final List<Path> coordinatePaths = chessPiece.getCoordinatePaths(position);
+        final HurdlePolicy hurdlePolicy = chessPiece.getHurdlePolicy();
+        return hurdlePolicy.pickDestinations(chessPiece.getTeam(), coordinatePaths, chessPiecePositions);
```

Comment:
hurdlePolicy를 두는 것은 괜찮은 시도인 것 같아요. 😃
하지만 잘 생각해보면 `hurdlePolicy`와 `coordinatePaths` 모두, chessPiece 객체가 상태로 가지거나 혹은 자신의 상태를 가지고 만들어낸 결과물로 보여요. (결국 getter를 통해 상태를 외부로 뽑아서 직접 처리하는 방식과 유사해지는 것 같네요 🤔)

그렇다면 chessPiece 객체에게 메시지를 보내서 chessPiece보고 수행해달라고 변경해볼 수 있을 것 같고 
그러면 결국 [AS IS](https://github.com/woowacourse/java-janggi/pull/40#issuecomment-2750561878)의 상태와 거의 똑같은 상태가 될 것 같아요. 

```suggestion
        return chessPiece.pickDestinations(position, chessPiecePositions);
```

저는 말씀하신 순환참조 문제를 해결하려면 Board 객체가 이동가능한 목적지를 직접 계산해주는 것도 괜찮을 것 같다고 생각해요. 🤔